VKappProject/VKappProject/Controllers/AvailableGroupsTableViewController.swift

```diff
@@ -34,33 +25,29 @@ final class AvailableGroupsTableViewController: UITableViewController {
 
     private var isSearching = false
 
-    // MARK: - Public Methods
-
-    func addGroup(_ group: [Group], complition: @escaping ((Group) -> Void)) {
-        availableGroups = availableGroups.filter { groups in
-            !group.contains { myGroup in
-                myGroup == groups
-            }
-        }
-        closureForGroup = complition
-    }
-
     // MARK: - Private Methods
 
-    private func chooseGroup(_ indexPath: IndexPath) {
-        let group = availableGroups[indexPath.row]
-        closureForGroup?(group)
-        navigationController?.popViewController(animated: true)
+    private func fetchSearchedGroupsRequest(searchedText: String) {
+        AF
+            .request(
+                "\(Constants.URLComponents.baseURL)" +
+                    "\(RequestType.searchGroups(searchQuery: searchedText).urlString)\(Constants.URLComponents.version)"
+            )
+            .responseData { response in
+                guard let data = response.value else { return }
+                do {
+                    let group = try JSONDecoder().decode(Group.self, from: data)
+                    self.searchResult = group.response.groups
+                } catch {
+                    print(error)
+                }
+            }
     }
 
     // MARK: - UITableViewDelegate, UITableViewDataSource
```

Comment:
```suggestion
    // MARK: - UITableViewDataSource
```
Зачем в марке UITableViewDelegate, если его методы ты не используешь?